lib/acts_as_taggable_on_mongoid/models/tag.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module ActsAsTaggableOnMongoid
+  module Models
+    # A class representing all tags that have ever been set on a model.
+    class Tag
```

Comment:
Moved from here:
https://github.com/mbleigh/acts-as-taggable-on/blob/master/lib/acts_as_taggable_on/tag.rb

Lots of changes for Mongoid.

Added context and taggable_type.  Otherewise tried to keep it mostly as-was.